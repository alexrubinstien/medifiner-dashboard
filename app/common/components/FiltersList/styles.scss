@import './app/common/styles/utils/_vars.scss';

.filters-list {
  margin: 7px 0 0;
  padding: 0;
  list-style: none;

  & & {
    margin-left: 31px;
  }

  &__item + &__item {
    margin-top: 7px;
  }

  &__filter {
    position: relative;
  }

  &__checkbox {
    position: absolute;
    top: 0;
    left: 0;
    width: 0;
    height: 0;
    opacity: 0;
  }

  &__fake-checkbox {
    position: relative;
    display: inline-block;
    width: 18px;
    height: 18px;
    background: $background-1;
    border: 2px solid $primary-color;
    border-radius: 2px;
    vertical-align: middle;

    &:before {
      position: relative;
      top: -1px;
    }
  }

  &__checkbox:checked + &__fake-checkbox {
    background: $primary-color;
    color: #fff;
    font-size: 10px;
    line-height: 18px;
    text-align: center;
  }

  &__checkbox:disabled + &__fake-checkbox {
    background: $background-3;
    border-color: $border-color;
  }

  &__filter-name {
    display: inline-block;
    margin-left: 8px;
    color: $text-color;
    font-size: 14px;
    font-weight: 500;
    line-height: 28px;
    vertical-align: middle;
  }

  &__checkbox:disabled ~ &__filter-name {
    opacity: .5;
    color: $text-light-color;
  }

  &__filter-count {
    float: right;
    color: #666F75;
    font-size: 14px;
    font-weight: 500;
    line-height: 28px;
  }

  &__checkbox:disabled ~ &__filter-count {
    opacity: .5;
    color: $text-light-color;
  }
}
