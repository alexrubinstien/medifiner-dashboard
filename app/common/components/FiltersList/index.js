import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import './styles.scss';

const FiltersList = ({
  filterName,
  filters,
  onChange,
}) => {
  return (
    <ul className="filters-list">
      {filters && filters.map(filter => (
        <li
          key={filterName+'-'+filter.id}
          className="filters-list__item"
        >
          <label
            htmlFor={filterName+'-'+filter.id}
            className="filters-list__filter"
          >
            <input
              type="checkbox"
              className="filters-list__checkbox"
              disabled={filter.disabled}
              id={filterName+'-'+filter.id}
              checked={filter.checked}
              onChange={() => onChange(filter, !filter.checked)}
            />
            <span
              className={classNames('filters-list__fake-checkbox', {
                'icon-tick': filter.checked,
              })}
            />
            <span className="filters-list__filter-name">
              {filter.name}
            </span>
            {!isNaN(filter.providersCount) && (
              <span className="filters-list__filter-count">
                ({filter.providersCount})
              </span>
            )}
          </label>
          <ul className="filters-list">
          {filter.organizations !== undefined && (
            filter.organizations.map(org => (
              <li
                key={filterName+'-'+filter.id+'-organizations-'+org.organizationId}
                className="filters-list__item"
              >
                <label
                  htmlFor={filterName+'-'+filter.id+'-organizations-'+org.organizationId}
                  className="filters-list__filter"
                >
                  <input
                    type="checkbox"
                    className="filters-list__checkbox"
                    disabled={org.disabled}
                    id={filterName+'-'+filter.id+'-organizations-'+org.organizationId}
                    checked={org.checked}
                    onChange={() => onChange(org, !org.checked, true)}
                  />
                  <span
                    className={classNames('filters-list__fake-checkbox', {
                      'icon-tick': org.checked,
                    })}
                  />
                  <span className="filters-list__filter-name">
                    {org.organizationName}
                  </span>
                  <span className="filters-list__filter-count">
                    ({org.providersCount})
                  </span>
                </label>
              </li>
            ))
          )}
          </ul>
        </li>
      ))}
    </ul>
  )
};

FiltersList.propTypes = {
  filterName: PropTypes.string.isRequired,
  filters: PropTypes.arrayOf(PropTypes.shape({
    checked: PropTypes.bool.isRequired,
    providersCount: PropTypes.number,
    name: PropTypes.string.isRequired,
    organizations: PropTypes.arrayOf(PropTypes.shape({
      checked: PropTypes.bool.isRequired,
      disabled: PropTypes.bool.isRequired,
      providersCount: PropTypes.number,
      organizationName: PropTypes.string.isRequired,
      organizationId: PropTypes.number.isRequired,
    })),
  })).isRequired,
  onChange: PropTypes.func.isRequired,
};

FiltersList.defaultProps = {
  parentFilterName: null,
};

export default FiltersList;
