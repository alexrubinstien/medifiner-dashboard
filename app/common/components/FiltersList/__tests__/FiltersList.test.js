/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import FiltersList from '../index.js';

describe('With Snapshot Testing', () => {
  it('Filters list component with flat filters', () => {
    const component = renderer.create(
      <FiltersList
        filters={[{
          checked: true,
          name: 'Oseltamivir75',
          title: 'Oseltamivir 75mg (pill)',
        }, {
          checked: true,
          name: 'Oseltamivir45',
          title: 'Oseltamivir 45mg (pill)',
        }, {
          checked: true,
          name: 'Oseltamivir30',
          title: 'Oseltamivir 30mg (pill)',
        }, {
          checked: true,
          name: 'oseltamivir',
          title: 'Oseltamivir Suspension (liquid)',
        }]}
        onChange={() => {}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Filters list component with nested filters', () => {
    const component = renderer.create(
      <FiltersList
        filters={[{
          checked: true,
          count: 528,
          name: 'chain',
          filters: [{
            checked: true,
            count: 120,
            name: 'cvs',
            title: 'CVS',
          }, {
            checked: true,
            count: 245,
            name: 'walmart',
            title: 'Walmart',
          }, {
            checked: true,
            count: 163,
            name: 'walgreens',
            title: 'Walgreens',
          }],
          title: 'Chain',
        }, {
          checked: true,
          count: 201,
          name: 'independent',
          title: 'Independent',
        }, {
          checked: true,
          count: 100,
          name: 'franchise',
          title: 'Franchise',
        }]}
        onChange={() => {}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
