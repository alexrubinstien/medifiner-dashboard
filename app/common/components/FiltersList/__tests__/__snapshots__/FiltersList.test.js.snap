// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Filters list component with flat filters 1`] = `
<ul
  className="filters-list"
>
  <li
    className="filters-list__item"
  >
    <label
      className="filters-list__filter"
      htmlFor="Oseltamivir75"
    >
      <input
        checked={true}
        className="filters-list__checkbox"
        disabled={false}
        id="Oseltamivir75"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="filters-list__fake-checkbox icon-tick"
      />
      <span
        className="filters-list__filter-name"
      >
        Oseltamivir 75mg (pill)
      </span>
    </label>
  </li>
  <li
    className="filters-list__item"
  >
    <label
      className="filters-list__filter"
      htmlFor="Oseltamivir45"
    >
      <input
        checked={true}
        className="filters-list__checkbox"
        disabled={false}
        id="Oseltamivir45"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="filters-list__fake-checkbox icon-tick"
      />
      <span
        className="filters-list__filter-name"
      >
        Oseltamivir 45mg (pill)
      </span>
    </label>
  </li>
  <li
    className="filters-list__item"
  >
    <label
      className="filters-list__filter"
      htmlFor="Oseltamivir30"
    >
      <input
        checked={true}
        className="filters-list__checkbox"
        disabled={false}
        id="Oseltamivir30"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="filters-list__fake-checkbox icon-tick"
      />
      <span
        className="filters-list__filter-name"
      >
        Oseltamivir 30mg (pill)
      </span>
    </label>
  </li>
  <li
    className="filters-list__item"
  >
    <label
      className="filters-list__filter"
      htmlFor="oseltamivir"
    >
      <input
        checked={true}
        className="filters-list__checkbox"
        disabled={false}
        id="oseltamivir"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="filters-list__fake-checkbox icon-tick"
      />
      <span
        className="filters-list__filter-name"
      >
        Oseltamivir Suspension (liquid)
      </span>
    </label>
  </li>
</ul>
`;

exports[`With Snapshot Testing Filters list component with nested filters 1`] = `
<ul
  className="filters-list"
>
  <li
    className="filters-list__item"
  >
    <label
      className="filters-list__filter"
      htmlFor="chain"
    >
      <input
        checked={true}
        className="filters-list__checkbox"
        disabled={false}
        id="chain"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="filters-list__fake-checkbox icon-tick"
      />
      <span
        className="filters-list__filter-name"
      >
        Chain
      </span>
      <span
        className="filters-list__filter-count"
      >
        (
        528
        )
      </span>
    </label>
    <ul
      className="filters-list"
    >
      <li
        className="filters-list__item"
      >
        <label
          className="filters-list__filter"
          htmlFor="cvs"
        >
          <input
            checked={true}
            className="filters-list__checkbox"
            disabled={false}
            id="cvs"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="filters-list__fake-checkbox icon-tick"
          />
          <span
            className="filters-list__filter-name"
          >
            CVS
          </span>
          <span
            className="filters-list__filter-count"
          >
            (
            120
            )
          </span>
        </label>
      </li>
      <li
        className="filters-list__item"
      >
        <label
          className="filters-list__filter"
          htmlFor="walmart"
        >
          <input
            checked={true}
            className="filters-list__checkbox"
            disabled={false}
            id="walmart"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="filters-list__fake-checkbox icon-tick"
          />
          <span
            className="filters-list__filter-name"
          >
            Walmart
          </span>
          <span
            className="filters-list__filter-count"
          >
            (
            245
            )
          </span>
        </label>
      </li>
      <li
        className="filters-list__item"
      >
        <label
          className="filters-list__filter"
          htmlFor="walgreens"
        >
          <input
            checked={true}
            className="filters-list__checkbox"
            disabled={false}
            id="walgreens"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="filters-list__fake-checkbox icon-tick"
          />
          <span
            className="filters-list__filter-name"
          >
            Walgreens
          </span>
          <span
            className="filters-list__filter-count"
          >
            (
            163
            )
          </span>
        </label>
      </li>
    </ul>
  </li>
  <li
    className="filters-list__item"
  >
    <label
      className="filters-list__filter"
      htmlFor="independent"
    >
      <input
        checked={true}
        className="filters-list__checkbox"
        disabled={false}
        id="independent"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="filters-list__fake-checkbox icon-tick"
      />
      <span
        className="filters-list__filter-name"
      >
        Independent
      </span>
      <span
        className="filters-list__filter-count"
      >
        (
        201
        )
      </span>
    </label>
  </li>
  <li
    className="filters-list__item"
  >
    <label
      className="filters-list__filter"
      htmlFor="franchise"
    >
      <input
        checked={true}
        className="filters-list__checkbox"
        disabled={false}
        id="franchise"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="filters-list__fake-checkbox icon-tick"
      />
      <span
        className="filters-list__filter-name"
      >
        Franchise
      </span>
      <span
        className="filters-list__filter-count"
      >
        (
        100
        )
      </span>
    </label>
  </li>
</ul>
`;
